feat(mls): navigate to migrated conversation [WPB-4705]

typfel opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4705" title="WPB-4705" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />WPB-4705</a>  [Android] Observe & navigate to active 1-1 conversation
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Re-opening https://github.com/wireapp/wire-android-reloaded/pull/2317 since it got due to a force push.

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

marcoconti83 commented:
> It would be good to make a Jira ticket with a link to this PR to help us not forget to retest these components.

I would prefer to rely on some code coverage report instead in the future. The coverage should tell us which classes/packages are not covered. 

The reason I am skeptical about creating a ticket is that one of tests I removed was one of the ~5 test suites in a package that has ~20 files that are meant to be tested. I'm pulling number from memory, but that's the ballpark figure. The coverage there is already pretty horrible, so I think there is little value in remembering that we need to cover one of those and not the 15 others. I suspect there are other sections of the app in a similar state.

vitorhugods committed:
feat(mls): redirect users to migrated
one-on-one conversation

This commit introduces a new ViewModel `ConversationMigrationViewModel.kt` for tracking conversation migrations. It observes details of a conversation and checks if this conversation was migrated to a different one-on-one conversation. If it was, it updates the target conversation with the ID of the active one-on-one conversation.

vitorhugods committed:
feat(mls): navigate to migrated conversation
